source/common/network/matching/inputs.h

```diff
@@ -32,9 +34,20 @@ class BaseFactory : public Matcher::DataInputFactory<MatchingDataType> {
 template <class MatchingDataType>
 class DestinationIPInput : public Matcher::DataInput<MatchingDataType> {
 public:
-  Matcher::DataInputGetResult get(const MatchingDataType& data) const override;
+  Matcher::DataInputGetResult get(const MatchingDataType& data) const override {
+    const auto& address = data.connectionInfoProvider().localAddress();
```

Comment:
I wonder if we could add a `data.localAddress()` to all the matching types and then we don't need the different template specializations? i.e make them all quack the same for these shared input types

Can defer this to another PR if you'd prefer, lmk

Reply:
Ok not important, we can merge for now